src/racer/ast.rs

```diff
@@ -636,6 +636,21 @@ impl<'c, 's, 'ast> visit::Visitor<'ast> for ExprTypeVisitor<'c, 's> {
                                 }
                                 Some(Ty::Match(m))
                             }
+                            MatchType::TypeParameter(ref traitbounds) if traitbounds.has_closure() => {
+                                let mut output = None;
+                                if let Some(path_search) = traitbounds.find_by_name("Fn")
```

Comment:
How about adding `TraitBounds::find_by_name(&self, names: &[&str]) -> Option<&PathSearch>`?

Reply:
I added a method `get_closure` but forgot to update this location.